### PR TITLE
Remove trailing space on "Forgot your password? "

### LINKED ISSUE #900: Forgot password needs "space" to be properly translated
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Internationalization

### Amplify Categories

auth

### Environment information

<details>

```
  System:
    OS: macOS 12.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 727.14 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.1 - ~/.volta/tools/image/node/14.17.1/bin/node
    Yarn: 1.22.11 - ~/.volta/tools/image/yarn/1.22.11/bin/yarn
    npm: 7.24.1 - ~/.volta/tools/image/npm/7.24.1/bin/npm
  Browsers:
    Chrome: 96.0.4664.55
    Firefox: 90.0.2
    Safari: 15.1
  npmPackages:
    @ampproject/toolbox-optimizer:  undefined ()
    @aws-amplify/ui-react: ^2.1.0 => 2.1.0
    @aws-amplify/ui-react-internal:  undefined ()
    @aws-amplify/ui-react-legacy:  undefined ()
    @babel/core:  undefined ()
    @peculiar/webcrypto:  undefined ()
    @types/node: 16.11.9 => 16.11.9
    @types/react: 17.0.36 => 17.0.36
    @vercel/nft:  undefined ()
    amphtml-validator:  undefined ()
    arg:  undefined ()
    async-retry:  undefined ()
    async-sema:  undefined ()
    aws-amplify: ^4.3.8 => 4.3.8
    babel-packages:  undefined ()
    ci-info:  undefined ()
    cli-select:  undefined ()
    comment-json:  undefined ()
    compression:  undefined ()
    conf:  undefined ()
    content-disposition:  undefined ()
    content-type:  undefined ()
    cookie:  undefined ()
    cross-spawn:  undefined ()
    debug:  undefined ()
    devalue:  undefined ()
    escape-string-regexp:  undefined ()
    eslint: 7.32.0 => 7.32.0
    eslint-config-next: 12.0.4 => 12.0.4
    find-cache-dir:  undefined ()
    find-up:  undefined ()
    formdata-node:  undefined ()
    fresh:  undefined ()
    glob:  undefined ()
    gzip-size:  undefined ()
    http-proxy:  undefined ()
    icss-utils:  undefined ()
    ignore-loader:  undefined ()
    is-animated:  undefined ()
    is-docker:  undefined ()
    is-wsl:  undefined ()
    json5:  undefined ()
    jsonwebtoken:  undefined ()
    loader-utils:  undefined ()
    lodash.curry:  undefined ()
    lru-cache:  undefined ()
    micromatch:  undefined ()
    mini-css-extract-plugin:  undefined ()
    nanoid:  undefined ()
    native-url:  undefined ()
    neo-async:  undefined ()
    next: 12.0.4 => 12.0.4
    node-libs-browser:  undefined ()
    ora:  undefined ()
    postcss-flexbugs-fixes:  undefined ()
    postcss-modules-extract-imports:  undefined ()
    postcss-modules-local-by-default:  undefined ()
    postcss-modules-scope:  undefined ()
    postcss-modules-values:  undefined ()
    postcss-preset-env:  undefined ()
    postcss-safe-parser:  undefined ()
    postcss-scss:  undefined ()
    postcss-value-parser:  undefined ()
    react: 17.0.2 => 17.0.2
    react-dom: 17.0.2 => 17.0.2
    react-server-dom-webpack:  undefined ()
    resolve-url-loader:  undefined ()
    sass-loader:  undefined ()
    schema-utils:  undefined ()
    semver:  undefined ()
    send:  undefined ()
    source-map:  undefined ()
    string-hash:  undefined ()
    strip-ansi:  undefined ()
    terser:  undefined ()
    text-table:  undefined ()
    typescript: 4.5.2 => 4.5.2
    ua-parser-js:  undefined ()
    unistore:  undefined ()
    uuid:  undefined ()
    web-streams-polyfill:  undefined ()
    web-vitals:  undefined ()
    webpack:  undefined ()
    webpack-sources:  undefined ()
    ws:  undefined ()
  npmGlobalPackages:
    npm: 6.14.13
```

</details>

### Describe the bug

According to docs to translate a string is just need to make a reference:

```ts
const dict = {
    'fr': {
        'Sign In': "Se connecter",
        'Sign Up': "S'inscrire"
    },
    'es': {
        'Sign In': "Registrarse",
        'Sign Up': "Regístrate"
    }
};

I18n.putVocabularies(dict);
```

But this doesn't work with "Forgot your password?" string in the `@aws-amplify/ui-react`

### Expected behavior

```ts
const dict = {
    'pt': {
        'Forgot your password?': "Esqueceu sua senha?",
    }
};

I18n.putVocabularies(dict);
```

### Reproduction steps

```ts
import { I18n } from 'aws-amplify';

// code

const dict = {
    'pt_br': {
        'Forgot your password?': "Esqueceu sua senha?",
    }
};

I18n.setLanguage('pt_br');
I18n.putVocabularies(dict);
```

But _works_ adding space after the original string as follows:

```ts
import { I18n } from 'aws-amplify';

// code

const dict = {
    'pt_br': {
        'Forgot your password? ': "Esqueceu sua senha?",
    }
};

I18n.setLanguage('pt_br');
I18n.putVocabularies(dict);
```

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

_No response_

### Manual configuration

_No response_

### Additional configuration

_No response_

### Mobile Device

_No response_

### Mobile Operating System

_No response_

### Mobile Browser

_No response_

### Mobile Browser Version

_No response_

### Additional information and screenshots

I checked the `@aws-amplify/ui-react` minified file and found out that is that way the string has the space:

Captura de Tela 2021-11-23 às 22 37 25

Looking for the code I asked myself if is not because of a {" "} put after the reference to string. But could be something in the build/bundle process.

I made a PoC here: https://github.com/ibrahimcesar/amplify-js/commit/8e2ce4caeb61e6edea39bfecdf5abb5bcc72921e#

Is better to have an uniform way to translate the strings and things like this space can waste some hours from devs!

### PR DESCRIPTION
**Pull request by wlee221**
*Issue #, if available:* Fixes #900.

*Description of changes:* Removes trailing space on "Forgot your password? ". This extra space was used in legacy authenticator to put spacing between "Forgot your password" and "Reset your password":

image

This is no longer needed or desired.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by slaymance — APPROVED** [review]
:raised_hands:

**Comment by wlee221** [comment]
Hmm, technically this is a breaking change so I'll try to add a backwards compatibility to this PR by today.

**Review by ErikCH — APPROVED** [review]
Looks, good but how is this linked to issue #990  ?

**Comment by wlee221** [comment]
@ErikCH Ahhhh typo, supposed to be #900 🤦

**Review by zchenwei — APPROVED** [review]
LGTM

**Comment by wlee221** [comment]
Added some backwards compatibility for legacy i18n key with trailing space. Will merge in 3 hours unless you guys find something.

File: packages/vue/src/components/sign-in.vue [hunk 1]
```diff
@@ -18,7 +23,10 @@ const emit = defineEmits([
 
 const passwordLabel = computed(() => translate('Password'));
 const forgotYourPasswordLink = computed(() =>
-  translate('Forgot your password? ')
+  // Support backwards compatibility for legacy key with trailing space
+  !hasTranslation('Forgot your password? ')
```
Comment (reesscot) [inline packages/vue/src/components/sign-in.vue:27]: Nice, thanks for supporting the old version of this

Comment (wlee221) [reply packages/vue/src/components/sign-in.vue:27]: Yep! We can remove this on our next major release.

**Pull request merged**